replica/table.cc

```diff
@@ -550,10 +555,28 @@ void table::enable_off_strategy_trigger() {
 
 storage_group_manager::~storage_group_manager() = default;
 
-future<> storage_group_manager::for_each_storage_group_gently(std::function<future<>(size_t, storage_group&)> f) {
-    rwlock::holder shared_lock = co_await get_rwlock().hold_read_lock();
-    for (auto& [id, sg]: _storage_groups) {
-        co_await f(id, *sg);
+// exception-less attempt to hold gate.
+// TODO: move it to seastar.
```

Comment:
Coming up in https://github.com/scylladb/seastar/commit/6acb4cfdd9624ae05f7564df3210955bf5dc96d1